addons/osfstorage/views.py

```diff
@@ -160,7 +161,7 @@ def osfstorage_create_child(file_node, payload, node_addon, **kwargs):
             created, file_node = True, parent.append_folder(name)
         else:
             created, file_node = True, parent.append_file(name)
-    except (KeyExistsException, ValidationError):
+    except (KeyExistsException, ValidationError, IntegrityError):
```

Comment:
Probably, my thought was preserving compatibility with toku but that's probably out the window as far as the addons directory goes.